Alignment/OfflineValidation/plugins/OverlapValidation.cc

```diff
@@ -383,7 +383,7 @@ void OverlapValidation::analyzeTrajectory(const Trajectory& trajectory,
     ++overlapCounts_[1];
     if ((layer != -1) && (acceptLayer[subDet])) {
       for (vector<TrajectoryMeasurement>::const_iterator itmCompare = itm - 1;
-           itmCompare >= measurements.begin() && itmCompare > itm - 4;
+           (itmCompare >= measurements.begin()) && ((itmCompare + 4) > itm);
```

Comment:
I'd argue comparing an iterator to be larger than or equal with `begin()` does not make much sense as in principle it is always `true` because 
> The begin iterator is not decrementable and the behavior is undefined if `--container.begin()` is evaluated.



https://en.cppreference.com/w/cpp/named_req/BidirectionalIterator